api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE, CREATING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), CREATING));
+        }
+        return validateResponse;
+    }
+
+    @Override
+    public AccountValidationResponse deleteAccount(String accountId) {
+        LOGGER.info(DELETING_ACCOUNT);
+        AccountValidationResponse response;
+        try {
+            BasicSessionCredentials credentials = credentialProvider.getBaseAccCredentials();
+            Regions region = Regions.fromName(System.getenv(REGION));
+            String roleName = System.getenv(PALADINCLOUD_RO);
+            AWSSecretsManager secretClient = AWSSecretsManagerClientBuilder
+                    .standard()
+                    .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                    .withRegion(region).build();
+            DeleteSecretRequest deleteRequest = new DeleteSecretRequest().withSecretId(secretManagerPrefix +
+                    FORWARD_SLASH + roleName + FORWARD_SLASH + serviceType() +
+                    FORWARD_SLASH + accountId).withForceDeleteWithoutRecovery(true);
+            DeleteSecretResult deleteResponse = secretClient.deleteSecret(deleteRequest);
+            LOGGER.info(DELETE_SECRET_RESPONSE, deleteResponse);
+            //delete entry from db
+            response = deleteAccountFromDB(accountId);
+            response.setType(Constants.REDHAT);
+            List<AccountDetails> onlineAccounts = findOnlineAccounts(STATUS_CONFIGURED, Constants.REDHAT);
+            if (onlineAccounts == null || onlineAccounts.isEmpty()) {
+                LOGGER.debug(ACCOUNTS_DELETED_MSG);
+                updateConfigProperty(REDHAT_ENABLED, FALSE, JOB_SCHEDULER);
+            }
+        } catch (Exception e) {
+            response = new AccountValidationResponse();
+            response.setValidationStatus(FAILURE);
+            response.setMessage(String.format(ERROR_MESSAGE_TEMPLATE, DELETING));
+            response.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), DELETING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE, DELETING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), DELETING));
+        }
+        return response;
+    }
+
+    private AccountValidationResponse validateRequest(CreateAccountRequest accountData) {
+        AccountValidationResponse response = new AccountValidationResponse();
+        StringBuilder validationErrorDetails = new StringBuilder();
+        if (accountData.getRedhatAccountId() == null || accountData.getRedhatAccountId().isEmpty()) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, PARAM_ACCOUNT_ID));
+        }
+
+        if (accountData.getRedhatAccountName() != null && accountData.getRedhatAccountName().trim().isEmpty()) {
```

Comment:
Are you assuming here that the UI will **always** send you a value here. Bare minimum that UI will send "". What if the UI sends null, wouldn't this if be skipped and validation would success if there were no other errors?

Reply:
Why even need the check for Account Name if this is an optional field? your createAccount will populate it with account id if it's null or empty.

Reply:
same code is used in validate API and add account API, so its better to check it has white spaces or not
So if we get some whitespaces in the place of account name, we will return error when Validate API hits